python/sglang/srt/constrained/future.py

```diff
@@ -0,0 +1,48 @@
+"""
+Copyright 2023-2024 SGLang Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""Base tool cache for constrained decoding tools."""
+
+import logging
+import threading
+from typing import Any, Callable
+
+logger = logging.getLogger(__name__)
+
+
+class FutureObject:
```

Comment:
I feel we do not need to re-implement the future. Can you use this https://docs.python.org/3/library/concurrent.futures.html#future-objects with ThreadPoolExecutor?

Reply:
Fixed in the latest commit!

But it brings a new problem: how to set up the default `max_workers` of a `ThreadPoolExecutor`?

Reply:
You can use a reasonable number such as 4 or 16.